### PR TITLE
Radius Proxy integration

### LINKED ISSUE #3773: Possibility to filter RADIUS accounting requests
In some cases, it could be necessary to filter RADIUS accounting requests coming from network devices.

It will be nice if we can do it with your filter engines (VLAN filters, RADIUS filters, etc.).

### PR DESCRIPTION
**Pull request by fdurand**
# Description
Configure in PacketFence RADIUS Proxy (acct and or auth) based on the realm of the user. (Easy way to define a remote radius server like NPS/Eduroam)
Call rest in the pre-proxy, post-proxy, preacct,  authorize and authorize in the tunnel section to use the radius filters.
Created a Dummy source just to compute the rules.
Call rest in post-auth for a specific realm (to compute a role and an access duration) 
Scope in the radius filters based on X-FreeRADIUS-Server and X-FreeRADIUS-Section (like packetfence.pre_proxy)

# Impacts
RADIUS workflow

# Delete branch after merge
YES

# Issue
fixes #3773 

# NEWS file entries
## New Features
* RADIUS Proxy integration from admin gui
* RADIUS Filter for pre_proxy/post_proxy/preacct/accounting/authorize
* DUMMY Source just to compute the rules

### CODE REVIEW

**Comment by fdurand** [comment]
Ready for review

File: debian/control [hunk 1]
```diff
@@ -35,7 +35,7 @@ Depends: ${misc:Depends}, vlan,
  libapache2-request-perl, libtie-dxhash-perl, libapache-session-perl,
  libapache-ssllookup-perl, libapache2-mod-systemd,
 # freeradius
- freeradius (>= 3.0.15.6), freeradius-ldap, 
+ freeradius (>= 3.0.18), freeradius-ldap,
```
Comment (nqb) [inline debian/control:38]: I didn't see Debian packages for freeradius 3.0.18

File: html/pfappserver/lib/pfappserver/Form/Config/Source/DUMMY.pm [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+
+=head1 NAME
+
+pfappserver::Form::Config::Source::DUMMY
+
+=cut
+
+=head1 DESCRIPTION
+
+Form definition to create or update an DUMMY authentication source.
```
Comment (nqb) [inline html/pfappserver/lib/pfappserver/Form/Config/Source/DUMMY.pm]: an DUMMY -> a DUMMY

**Comment by MuriloCalegari** [comment]
Hi @fdurand,

We're super excited to have a proper eduroam wifi integrated with Packetfence here at our school. Do you have a date for when this RADIUS proxy will be available?

**Comment by fdurand** [comment]
Hi MuriloCalegari,
the code works, i just need to do some small change.
It will be part of the 8.3 release (near the new the year).

File: conf/pf.conf.defaults [hunk 3]
```diff
@@ -1100,22 +1076,82 @@ user=
...
 # EAP-FAST authority ID (Considered only if EAP-FAST is enabled)
 eap_fast_authority_identity=1234
+#
+# radius_configuration.record_accounting_in_sql
+#
+# Record the accounting data in the SQL tables.
+# Requires a restart of radiusd to be effective.
+record_accounting_in_sql=enabled
+#
+# advanced.filter_in_authorize
```
Comment (jrouzierinverse) [inline conf/pf.conf.defaults]: Change the name
advanced.* -> radius_configuration.*

**Comment by extrafu** [comment]
@jrouzierinverse What's the status on this?

**Comment by jrouzierinverse** [comment]
rebase

**Comment by fdurand** [comment]
Done

**Comment by jrouzierinverse** [comment]
rebase again

**Comment by fdurand** [comment]
done

**Comment by jrouzierinverse** [comment]
Looks good

**Comment by fdurand** [comment]
Can you merge it ?

File: conf/pf.conf.defaults [hunk 4]
```diff
@@ -1090,22 +1066,82 @@ user=
...
+# Requires a restart of radiusd to be effective.
+filter_in_packetfence_authorize=disabled
+#
+# advanced.filter_in_pre_proxy
+#
+# Send the radius request in the radius filter from the radius packetfence.pre_proxy section.
+# Requires a restart of radiusd to be effective.
+filter_in_packetfence_pre_proxy=disabled
+#
+# advanced.filter_in_post_proxy
```
Comment (jrouzierinverse) [inline conf/pf.conf.defaults]: advanced.* -> radius_configuration.*

File: conf/pf.conf.defaults [hunk 5]
```diff
@@ -1090,22 +1066,82 @@ user=
...
+# Requires a restart of radiusd to be effective.
+record_accounting_in_sql=enabled
+#
+# advanced.filter_in_authorize
+#
+# Send the radius request in the radius filter from the radius packetfence.authorize section.
+# Requires a restart of radiusd to be effective.
+filter_in_packetfence_authorize=disabled
+#
+# advanced.filter_in_pre_proxy
```
Comment (jrouzierinverse) [inline conf/pf.conf.defaults]: advanced.* -> radius_configuration.*

**Review by jrouzierinverse — CHANGES_REQUESTED** [review]
Create an upgrade script to move old configuration to the new location

**Comment by fdurand** [comment]
Change done

**Comment by julsemaan** [comment]
Just a note so you guys don't forget the upgrade notes for the script @jrouzierinverse created

**Pull request merged**